impl/src/main/java/org/ehcache/internal/classes/ClassInstanceProvider.java

```diff
@@ -68,15 +68,27 @@ protected T newInstance(String alias, ServiceConfiguration<?> serviceConfigurati
     return newInstance(alias, config);
   }
 
-  private T newInstance(String alias, ClassInstanceConfiguration<T> config) {
+  protected T newInstance(String alias, Object ... args) {
+    return newInstance(alias, null, args);
+  }
+
+  private T newInstance(String alias, ClassInstanceConfiguration<T> config, Object... args) {
     if (config == null) {
       config = getPreconfigured(alias);
       if (config == null) {
         return null;
       }
     }
+    final Object[] constructorArgs = args.length > 0 ? args : config.getArguments();
     try {
-      return invokeConstructor(config.getClazz(), config.getArguments());
+      try {
+        return invokeConstructor(config.getClazz(), constructorArgs);
+      } catch (NoSuchMethodException e) {
```

Comment:
This is getting seriously ugly... Should have such fallback? 
Or maybe I should try to merge the args in the config with the `Serializer` one ? Or override them if it's a `SerializingCopier` we're instantiating for that `Store`? I _think_ this could then all be done in the `DefaultCopyProvider` class... Hopefully! Anyways, will sleep over that one.